compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt

```diff
@@ -276,13 +285,29 @@ internal actual class ComposeWindow : UIViewController {
             interfaceOrientationState.value = it
         }
 
-        val composeLayer = attachedComposeContext?.composeLayer ?: return
+        attachedComposeContext?.let {
+            updateLayout(it)
+        }
+    }
 
-        val (width, height) = getViewFrameSize()
-        val scale = density.density
+    private fun updateLayout(context: AttachedComposeContext) {
+        context.scene.also {
+            it.density = density
+            it.constraints = view.frame.useContents {
+                val scale = density.density
+
+                val pixelsFromDps = { value: Double ->
+                    (value * scale.toDouble()).roundToInt()
+                }
 
-        composeLayer.setDensity(density)
-        composeLayer.setSize((width * scale).roundToInt(), (height * scale).roundToInt())
+                Constraints(
+                    maxWidth = pixelsFromDps(size.width),
+                    maxHeight = pixelsFromDps(size.height)
```

Comment:
This logic is very simple and maybe easier to just simplify:
```suggestion
                    maxWidth = (scale * size.width).roundToInt(),
                    maxHeight = (scale * size.height).roundToInt()
```

Reply:
With explicit casts removed it is indeed, updated